Common/Extensions/StreamExtensions.cs

```diff
@@ -1057,37 +1057,33 @@ public static byte[] ReadDelimitedValue(this System.IO.Stream stream, byte delim
             //The result of reading from the stream
             byte[] result = null;
 
-            try
-            {
-                //Declare a value which will end up in a register on the stack
-                int register = -1;
+            //Declare a value which will end up in a register on the stack
+            int register = -1;
 
-                //Indicate when to terminate reading.
-                bool terminate = false;
+            //Indicate when to terminate reading.
+            bool terminate = false;
 
-                //Use a MemoryStream as to not lock the reader
-                using (var buffer = new System.IO.MemoryStream())
+            //Use a MemoryStream as to not lock the reader
+            using (var buffer = new System.IO.MemoryStream())
+            {
+                //While data can be read from the stream
+                while (false == terminate)
                 {
-                    //While data can be read from the stream
-                    while (false == terminate)
-                    {
-                        //Read a byte from the stream
-                        register = stream.ReadByte();
+                    //Read a byte from the stream
+                    register = stream.ReadByte();
 
-                        //Check for termination
-                        terminate = register == -1 || register == delimit;
+                    //Check for termination
+                    terminate = register == -1 || register == delimit;
 
-                        //If the byte read is equal to the delimit and the delimit byte is not included then return the array contained in the MemoryStream.
-                        if (terminate && false == includeDelimit) break;
+                    //If the byte read is equal to the delimit and the delimit byte is not included then return the array contained in the MemoryStream.
+                    if (terminate && false == includeDelimit) break;
 
-                        //Write the value read from the reader to the MemoryStream
-                        buffer.WriteByte((byte)register);
-                    }
-                    //If terminating then return the array contained in the MemoryStream.
-                    result = buffer.ToArray();
+                    //Write the value read from the reader to the MemoryStream
+                    buffer.WriteByte((byte)register);
                 }
+                //If terminating then return the array contained in the MemoryStream.
+                result = buffer.ToArray();
```

Comment:
I wonder if we can use MemorySegment here or Span or some other way to reduce the allocation here.

Reply:
```

 public static Common.MemorySegment ReadDelimitedValue(this System.IO.Stream stream, byte delimit = Common.ASCII.LineFeed, bool includeDelimit = false)
        {
            try
            {
                //Declare a value which will end up in a register on the stack
                int register = -1;

                //Indicate when to terminate reading.
                bool terminate = false;

                //Use a MemoryStream as to not lock the reader
                using (var buffer = new System.IO.MemoryStream())
                {
                    //While data can be read from the stream
                    while (false == terminate)
                    {
                        //Read a byte from the stream
                        register = stream.ReadByte();

                        //Check for termination
                        terminate = register == -1 || register == delimit;

                        //If the byte read is equal to the delimit and the delimit byte is not included then return the array contained in the MemoryStream.
                        if (terminate && false == includeDelimit) break;

                        //Write the value read from the reader to the MemoryStream
                        buffer.WriteByte((byte)register);
                    }
                    //If terminating then return the array contained in the MemoryStream.
                    var result = buffer.ToArray();

                    //Return the bytes read from the stream
                    return new Common.MemorySegment(result, 0, (int)buffer.Position);
                }
            }
            catch { throw; }            
        }
        
        ```